### PR TITLE
Adds realsense d415 urdf and mesh

### PR DESCRIPTION
**Pull request by IanTheEngineer**
This PR adds the code to fetch the models repo d415 mesh to Drake along with a simple URDF. The Realsense d415 mesh and Apache 2.0 license from https://github.com/IntelRealSense/realsense-ros `realsense2_description` package.
- The tests are closely based (copied + slightly modified) on the YCB meshes simple parsing tests.
- This PR depends on https://github.com/RobotLocomotion/models/pull/14
- @RussTedrake requested this mesh for the current Manipulation course

d415_drake_visualizer

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080)

### CODE REVIEW

**Review by IanTheEngineer — COMMENTED** [review]
+@jwnimmer-tri  for review please

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080#-:-MHNcfoM5nPBK8YQyta-:bnw12vd)* status: LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @jwnimmer-tri)

**Comment by jwnimmer-tri** [comment]
> jwnimmer-tri for Platform review please.

The first step is feature review, not platform review.  I can do the feature review; we'll use the next on-call platform reviewer once feature review is complete.

> RussTedrake requested this mesh for the current Manipulation course.

Just to clarify -- is the plan for code inside drake itself to start referring to this mesh?  For example, the manipulation_station will use it (and cite it in the BUILD file, etc.)?  If the _only_ use is the manipulation course repo, then the models should be committed there.  If a Drake example will use it, it's fine to commit here.

**Review by IanTheEngineer — COMMENTED** [review]
Ah, I mispoke and was just in the midst of correcting. I intended for you to do feature review. 

As for whether this is for the course only or an addition to the manipulation_station, I am not certain, but perhaps @russtedrake could clarify. I could definitely see the realsense cameras being added to the manipulation_station example.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080#-:-MHNjVGVArYeyZ5yrcGI:b-v0ds2o)* status: LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @jwnimmer-tri)

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080)* status: LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @jwnimmer-tri)

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080)* status: 1 unresolved discussion, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @IanTheEngineer and @jwnimmer-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/14080#-MHNlKl_0MD2f8UBnrAi:-MHNlKl_0MD2f8UBnrAj:b-n9zk3z) (no related file):*
Second question (as new reviewable thread):

Did you have to _edit_ the mesh file(s) at all vs upstream?  If not, we could (and probably should?) just declare the realsense2_description as a github dependency and fetch it a build time, instead of copying it its data into the models repository.

---

**Review by IanTheEngineer — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080)* status: 1 unresolved discussion, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @jwnimmer-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/14080#-MHNlKl_0MD2f8UBnrAi:-MHNmHGR7iqyD_6uarMG:bsbau09) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Second question (as new reviewable thread):

Did you have to _edit_ the mesh file(s) at all vs upstream?  If not, we could (and probably should?) just declare the realsense2_description as a github dependency and fetch it a build time, instead of copying it its data into the models repository.
</blockquote></details>

I didn't have to edit the mesh file at all, but the Xacro file had two issues:
1) it was missing a default argument that was required to construct the URDF
2) the interaction between Bazel and xacro-python was unhappy about dependencies. See the following gist: https://gist.github.com/IanTheEngineer/24b6b00e4aa5e5e76d13042236ab565e

As a result, I just preconstructed the urdf and reference the unmodified meshes.

---

**Review by RussTedrake — COMMENTED** [review]
Thanks!  I plan to add it to the manipulation station, and also use it in many small python examples that involve the cameras and objects (but not the entire station).

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080#-:-MHNtpjOC2FIoZKT1R8M:bwic4ic)* status: 1 unresolved discussion, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @jwnimmer-tri)

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080)* status: 2 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @IanTheEngineer and @jwnimmer-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/14080#-MHNlKl_0MD2f8UBnrAi:-MHOGHfF5nc9roc4nhhk:b-wet92w) (no related file):*
<details><summary><i>Previously, IanTheEngineer (Ian McMahon) wrote…</i></summary><blockquote>

I didn't have to edit the mesh file at all, but the Xacro file had two issues:
1) it was missing a default argument that was required to construct the URDF
2) the interaction between Bazel and xacro-python was unhappy about dependencies. See the following gist: https://gist.github.com/IanTheEngineer/24b6b00e4aa5e5e76d13042236ab565e

As a result, I just preconstructed the urdf and reference the unmodified meshes.
</blockquote></details>

In that case, since upstream is quite active, and it seems plausible that we'll want more meshes of camera models in the future, how about adding upstream's github to our workspace and using it for the meshes, instead of RobotLocomotion/models?

---
___
*[manipulation/models/realsense2_description/LICENSE, line 1 at r1](https://reviewable.io/reviews/robotlocomotion/drake/14080#-MHNmRHM48gs1nMWYf2h:-MHNmRHM48gs1nMWYf2i:b-h2c895) ([raw file](https://github.com/robotlocomotion/drake/blob/0563fd517639cbbbd9492d967b98a2aff16c208c/manipulation/models/realsense2_description/LICENSE#L1)):*
> ```
> 
> ```

It is unclear to which files in this directory this license applies.  At minimum, we'd place a README.md that answers it.  (We should introduce a better practice in the future, but for now it's the convention.)

For files that are copied from a different location, we also ask that the commit message and/or README describe the original source URL and the date or revision the copy was made from.

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14080)* status: 2 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers, commits need curation (https://drake.mit.edu/reviewable.html#curated-commits) (waiting on @IanTheEngineer and @jwnimmer-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/14080#-MHNlKl_0MD2f8UBnrAi:-MHOlZSi7gmRaI-Sbhd6:b-bgwr7t) (no related file):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

In that case, since upstream is quite active, and it seems plausible that we'll want more meshes of camera models in the future, how about adding upstream's github to our workspace and using it for the meshes, instead of RobotLocomotion/models?
</blockquote></details>

I also looked at the xacro challenges.

I don't think the "test_d415" file is super useful.  We can re-implement ourselves (and thus customize exactly how we've invoking the helper macros), so I think that solves (1).

For (2), the dependencies are only onto the self-same package (i.e., no-ops), so I think we can just strip them out.  A `sed -i -e 's|$(find realsense2_description)/urdf/||' *.xacro` once the files have been fetched from upstream seems to solve it.  (We can run `sed` using a `genrule()` in bazel, or possibly one `genrule()` each per file in a loop.)

In short, I'd like us to give a stab at using upstream for everything interesting, and just write a BUILD file to make it happen.  I can help with any nuances there.

https://github.com/RobotLocomotion/drake/tree/master/tools/workspace/#adding-new-third-party-software

Using upstream will make it super easy to pull in bugfixes and new camera models over time, and reduces our workload for copying vendored files around.  (The other models we keep forked inside Drake are because either upstream is dead and we are doing our own bugfixes, or because we needed to entirely rewrite the collision geometry, or etc.)

---

**Comment by IanTheEngineer** [comment]
Closed in favor of using the repo directly in https://github.com/robotlocomotion/drake/pull/14203
